sqlite.go

```diff
@@ -490,6 +490,14 @@ func (s *stmt) ExecContext(ctx context.Context, args []driver.NamedValue) (drive
 	if err := s.bindAll(args); err != nil {
 		return nil, s.reserr("Stmt.Exec(Bind)", err)
 	}
+	if ctx.Value(queryCancelKey{}) != nil {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithCancel(ctx)
+		defer cancel()
+
+		db := s.stmt.DBHandle()
+		go func() { <-ctx.Done(); db.Interrupt() }()
```

Comment:
It might be worth only calling `Interrupt` if the statement is still running, which you could check with `s.bound` I believe.

Reply:
(same issue as noted below)